src/shims/tls.rs

```diff
@@ -221,12 +277,13 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             this.run()?;
 
             // Fetch next dtor after `key`.
-            dtor = match this.machine.tls.fetch_tls_dtor(Some(key)) {
+            dtor = match this.machine.tls.fetch_tls_dtor(Some(key), thread_id) {
                 dtor @ Some(_) => dtor,
                 // We ran each dtor once, start over from the beginning.
-                None => this.machine.tls.fetch_tls_dtor(None),
+                None => this.machine.tls.fetch_tls_dtor(None, thread_id),
             };
         }
+
```

Comment:
Should we "clean up" here, and remove this thread from all the maps of all TLS vars? Just to avoid accumulating cruft in the data structure here.